src/docs/asciidoc/hazelcast_clients.adoc

```diff
@@ -1288,23 +1287,24 @@ Its default value is 1000 ms.
 * `max-backoff-millis`: Specifies the upper limit for the backoff in milliseconds.
 Its default value is 30000 ms.
 * `multiplier`: Factor to multiply the backoff after a failed retry.
-Its default value is 2.
-* `fail-on-max-backoff`: Specifies whether to fail when the `max-backoff-millis` has reached or
-continue waiting `max-backoff-millis` at each iteration. Its default value is false.
-* `jitter`: Specifies by how much to randomize backoffs. Its default value is 0.2.
+Its default value is 1.
+* `cluster-connect-timeout-millis`: Timeout value in milliseconds for the client to give up to connect to the current cluster
+Its default value is 20000.
+* `jitter`: Specifies by how much to randomize backoffs. Its default value is 0.
 
 A pseudo-code is as follows:
 
 [source,plain]
 ----
+ begin_time = getCurrentTime()
  current_backoff_millis = INITIAL_BACKOFF_MILLIS
- while (TryConnect(connectionTimeout)) != SUCCESS)
-    if (fail-on-max-backoff && current_backoff_millis >= MAX_BACKOFF_MILLIS) {
-        FAIL WITH EXCEPTION
+ while (TryConnect(connectionTimeout)) != SUCCESS) {
+    if (getCurrentTime() - begin_time >= CLUSTER_CONNECT_TIMEOUT_MILLIS) {
+        GIVE UP
```

Comment:
The previous doc `FAIL WITH EXCEPTION` may be better.

Reply:
In failover scenerio, it does not fail with exception but only switch to next cluster. I tried to make the code consistent with the `CLUSTER_CONNECT_TIMEOUT_MILLIS` explanation above.

Reply:
If it will be consistent with `CLUSTER_CONNECT_TIMEOUT_MILLIS `, you can better say `Give up to connecting to the current cluster and switch to another if exists` . Just saying `GIVE UP` may mean a lot different things.